### ISSUE TITLE
Change admin password

### ISSUE THREAD

**Issue by ronaldvaneede** [bstopp/puppet-aem#49]
It would be nice if it is possible to change the default admin password to something else.

**Comment by cliffano** [comment]
@rveede If you still need the ability to change admin/user password, check out https://github.com/shinesolutions/puppet-aem-resources .

It's designed to complement puppet-aem , it has a number of features for managing AEM resources, but for changing user password in particular, here's a sample snippet:

```
aem_user { 'Change admin default password':
  ensure       => password_changed,
  name         => 'admin',
  path         => '/home/users/a',
  old_password => 'admin',
  new_password => 'newadminpassword'
}
```

**Comment by rveede** [comment]
@cliffano I am no longer working on the project I used this aem module for. It looks nice though. I am only wondering if that works with AEM 6.1(?) and up because afaik the user content paths are hashed since 6.0 or 6.1.

**Comment by cliffano** [comment]
@rveede Yes, it is heavily used for a number of AEM 6.2 projects and works with 6.1 too.

The hashed paths are called authorizable IDs and they were introduced in AEM 6.1 .
It is handled on the underlying ruby_aem layer https://github.com/shinesolutions/ruby_aem .

Both user and group resources within ruby_aem provide `find_authorizable_id` method for mapping a user/group name to the corresponding authorizable ID, hence allowing the client code to not worry about the authorizable ID, and instead only use the user/group name, e.g. 'admin', 'alice', 'bob', etc.

**Comment by bdhoine** [comment]
It is possible to provision the initial password since AEM 6.3 with better support towards 6.5. The aproach decribed in the docs could be used to achieve this (https://helpx.adobe.com/experience-manager/6-5/sites/administering/using/security-configure-admin-password.html).
